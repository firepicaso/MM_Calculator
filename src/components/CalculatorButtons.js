import React from 'react';
import PropTypes from 'prop-types';

function CalculatorButtons({ label, type, onClick }) {
  return (
    <button type="button" className={type} onClick={onClick}>
      {label}
    </button>
  );
}

CalculatorButtons.propTypes = {
  label: PropTypes.string.isRequired,
  type: PropTypes.string,
  onClick: PropTypes.func,
};

CalculatorButtons.defaultProps = {
  type: '',
  onClick: () => {},
};

export default CalculatorButtons;
