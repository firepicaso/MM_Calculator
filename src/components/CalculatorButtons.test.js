import CalculatorButtons from "./CalculatorButtons";
import { render, screen, fireEvent } from '@testing-library/react';

test('renders Button component', () => {
    const { container } = render(<CalculatorButtons label="1" onClick={() => {}} />);
    expect(container.firstChild).toMatchSnapshot();
  });
  
  test('simulating user interaction', () => {
    const clickButtons = jest.fn();
    render(<CalculatorButtons label="1" onClick={clickButtons} />);
    const buttonElement = screen.getByText('1');
    fireEvent.click(buttonElement);
    fireEvent.click(buttonElement);
    expect(clickButtons).toHaveBeenCalledTimes(2);
});
