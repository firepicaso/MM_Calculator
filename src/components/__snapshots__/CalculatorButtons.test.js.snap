// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing Button component 1`] = `
<button
  class=""
  type="button"
>
  1
</button>
`;
